### ISSUE TITLE
Zed detects incorrect Python version (3.13 instead of 3.12) for virtual environments

### ISSUE THREAD

**Issue by purejoy** [zed-industries/zed#20390]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Description:
When opening a Python project in Zed, it correctly detects the virtual environments I have created locally, but it detects the wrong Python version. I created virtual environments with Python 3.12, but Zed detects them as Python 3.13.However, this issue does not occur with the system-wide Python environment, which Zed detects correctly.

Steps to Reproduce:
1.Create a virtual environment using Python 3.12 (e.g., mkvirtualenv or uv).
2.Open the Python project in Zed.
3.Check the detected Python version for the virtual environment.
4.Observe that Zed detects the virtual environment as Python 3.13 instead of 3.12.

### Environment

Zed: v0.160.7 (Zed)
OS: Linux Wayland gentoo 2.17
Memory: 15.6 GiB
Architecture: x86_64
GPU: AMD Radeon RX 560 Series (RADV POLARIS11) || radv || Mesa 24.2.6

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/user-attachments/assets/a66013f8-4c60-4d97-9880-2bc404d9e140)

### If applicable, attach your Zed.log file to this issue.

[zed.log](https://github.com/user-attachments/files/17671841/zed.log)

**Comment by alatecj** [comment]
Same issue here, but 3.9.20 detected instead of 3.13.0

### Describe the bug / provide steps to reproduce it

1. created venv with python -m venv .venv
2. activated with source .venv/bin/activate
3. python -V shows 3.13.0
4. after running Zed, Python 3.9.20 is detected

### Environment

Zed: 0.160.7
OS: Fedora 41
Architecture: x86_64
GPU: "AMD Radeon RX 6700 XT (RADV NAVI22)", driver_name: "radv", driver_info: "Mesa 24.2.5"

### If applicable, add mockups / screenshots to help explain present your vision of the feature
![image](https://github.com/user-attachments/assets/f8513d27-7fea-40b5-919a-7f8ad146e01a)

If applicable, attach your Zed.log file to this issue.
[Zed.log](https://github.com/user-attachments/files/17676776/Zed.log)

**Comment by osiewicz** [comment]
Can you repro this issue when running https://github.com/microsoft/python-environment-tools directly from the root of a project?

**Comment by purejoy** [comment]
> Can you repro this issue when running https://github.com/microsoft/python-environment-tools directly from the root of a project?

Yes, the same issue occurred. pet cannot correctly identify the Python version of the virtual environment
![image](https://github.com/user-attachments/assets/b522ca92-818c-4e88-bcb7-d3e7e6fc5fa1)

**Comment by scottpashley** [comment]
I'm in the same situation as alatecj, the only difference is my environment was created by Poetry rather than venv
